datajunction-server/datajunction_server/api/nodes.py

```diff
@@ -1486,13 +1486,14 @@ def tag_a_node(
     )
 
 
-def create_new_revision_from_existing(  # pylint: disable=too-many-locals,too-many-arguments
+def create_new_revision_from_existing(  # pylint: disable=too-many-locals,too-many-arguments,too-many-branches
     session: Session,
     query_service_client: QueryServiceClient,
     old_revision: NodeRevision,
     node: Node,
     data: UpdateNode = None,
     version_upgrade: VersionUpgrade = None,
+    refresh_if_source: bool = True,
```

Comment:
Since there are longer term plans for reflection to do other things like update materialization information, valid through timestamps, and maybe distinct values for dimensions, would this arg be better as just `refresh: bool`?

Reply:
Makes sense. Let me update this.

Reply:
@agorajek Yeah, good point. I think the only complications are around conflict resolution between the custom columns and the actual columns. The logic here for source nodes would be:
* Retrieve latest columns from query service
* If there are any custom columns passed in, filter down to only those custom columns or raise an error if the custom ones don't match the ones from the query service
* If there are no custom columns, just update the source node's columns to the latest from the query service
* Propagate any dimension links defined on the old source node revision -> it looks like the custom columns allow for adding dimension links, but I don't think we should allow that because it adds to the confusion
* Propagate any primary keys defined on the old source node revision OR update to new primary keys if provided in the update config

Reply:
@shangyian I am a bit puzzled by the "custom columns" term you used. Maybe we can try to chat briefly today to clarify?

Reply:
@agorajek I meant that when you create a source node, you can either provide the column names + types or rely on the query service to tell you what they are

Reply:
I see, yeah, I don't see much value in the user telling the system how the source nodes look like.